torch/nn/utils/_named_member_accessor.py

```diff
@@ -128,6 +151,15 @@ def get_submodule(self, name: str) -> "torch.nn.Module":
             self.memo[name] = submodule
             return submodule
 
+    def swap_submodule(self, path: str, value: "torch.nn.Module") -> "torch.nn.Module":
+        """
+        Swap the submodule specified by the given ``path`` to ``value``.
+        For example, to swap the attribute mod.layer1.conv1 use
+        ``accessor.swap_submodule("layer1.conv1", conv2)``.
+        """
+        prefix, _, attr = path.rpartition(".")
+        return swap_submodule(self.get_submodule(prefix), attr, value)
```

Comment:
Swapping submodules may break the caching mechanism in `NamedMemberAccessor`. If we want this works as expected, the batched version of `swap_submodule` should handle the order carefully.

```python
accessor.swap_submodule('layer1.conv1', myconv1)
accessor.swap_submodule('layer1', mylayer1)
```

```python
accessor.swap_submodule('layer1', mylayer1)
accessor.swap_submodule('layer1.conv1', myconv1)
```